.github/workflows/_shared-first-time-setup.yml

```
@@ -0,0 +1,50 @@
+name: "Repository First Time Setup automation (shared)"
+
+on:
+  workflow_call:
+    inputs:
+        esf_version:
+          type: string
+          default: 'https://changeme'
+          required: true
+        groupId:
+          type: string
+          default: ''
+          required: true
+        artifactId:
+          type: string
+          default: ''
+          required: true
+
+jobs:
+  init:
+    runs-on: ubuntu-latest
+    steps:
+
+    - name: Download ESF archetype
+      uses: carlosperate/download-file-action@v1
+      with:
+        file-url: ${{ github.event.inputs.esf_version }}
+      if: startsWith(${{ github.event.inputs.esf_version }}, 'https://eth-repo.s3.amazonaws.com/esf')
```

The if statement is not being run. Might be because it needs to be moved after the `name`.

```suggestion
    - name: Download ESF archetype
      if: startsWith(${{ github.event.inputs.esf_version }}, 'https://eth-repo.s3.amazonaws.com/esf')
      uses: carlosperate/download-file-action@v1
      with:
        file-url: ${{ github.event.inputs.esf_version }}
```